src/App.js

```diff
@@ -1,16 +1,85 @@
 import React from "react";
 import logo from "./logo.png";
 import "./App.css";
+import axios from "axios";
 
 class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      city: "",
+      weather_description: "",
+      temperature: "",
+      humidity: "",
+      error: "", // Add error state
+    };
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const APIKey = process.env.REACT_APP_API_KEY;
```

Comment:
I would place this outside of the component, so it is always available and not has to be looked up on every form submission.